pallets/schemas/src/types.rs

```diff
@@ -18,3 +39,56 @@ pub struct SchemaInfo {
 	/// additional control settings for the schema
 	pub settings: SchemaSettings,
 }
+
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+/// A structure defining name of a schema
+pub struct SchemaName {
+	/// namespace or domain of the schema
+	pub namespace: BoundedVec<u8, ConstU32<NAMESPACE_MAX>>,
+	/// name or descriptor of this schema
+	pub descriptor: BoundedVec<u8, ConstU32<DESCRIPTOR_MAX>>,
+}
+
+impl SchemaName {
+	/// parses and verifies the request and returns the SchemaName type if successful
+	pub fn try_parse<T: Config>(payload: SchemaNamePayload) -> Result<SchemaName, DispatchError> {
```

Comment:
I think we could just limit the schema name parts to just lowercase `a-z` and `-`. While we can fix stuff for the submitter, as this comes through governance we can just force the input to be correct